### ISSUE TITLE
False-positive NullableDereference raised for anonymous class referencing this of enclosed class

### ISSUE THREAD

**Issue by pzygielo** [google/error-prone#1508]
### Description of the problem:

False-positive<sup>*</sup> `NullableDereference` raised for anonymous class referencing `this` of enclosed class.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.
error-prone compile (with `NullableDereference` enabled) following piece:
```java
package ep;

import java.util.concurrent.Callable;

public interface Issue {

    default public Callable<String> getCallable() {
        return new Callable<String>() {
            @Override
            public String call() {
                return Issue.this.toString();
            }
        };
    }
}
```
to get
```
Issue.java:[11,24] [NullableDereference] Dereferencing method/field "this" of possibly null receiver Issue
    (see https://errorprone.info/bugpattern/NullableDereference)
```

### What version of Error Prone are you using?
2.3.4

###  Have you found anything relevant by searching the web?

> - GitHub issues: https://github.com/google/error-prone/issues

It's experimental check disabled by default.

> - GoogleGroups

https://groups.google.com/forum/#!searchin/error-prone-discuss/NullableDereference%7Csort:date: no results

---
<sup>*</sup> - I can't think of the way this can really happen.
